libsolidity/ast/Types.cpp

```diff
@@ -2050,6 +2050,8 @@ unsigned StructType::calldataOffsetOfMember(std::string const& _member) const
 
 bool StructType::isDynamicallyEncoded() const
 {
+	if (recursive())
+		return true;
```

Comment:
A lovely easy fix :)

Reply:
Well, but not the best one and not a permanent one given https://github.com/ethereum/solidity/issues/7498 ;-).